dhctl/pkg/template/bundle.go

```diff
@@ -184,3 +189,57 @@ func withoutNodeGroup(data map[string]interface{}) map[string]interface{} {
 	}
 	return filteredData
 }
+
+func RenderAndSaveDetectBundle(data map[string]interface{}) (string, error) {
+	log.DebugLn("Start render detect bundle script")
+
+	fileContent, err := ioutil.ReadFile(detectBundlePath)
+	if err != nil {
+		return "", fmt.Errorf("loading %s: %v", detectBundlePath, err)
+	}
+
+	content := string(fileContent)
+
+	if data != nil {
+		t := template.New("detect_bundle_render").Funcs(FuncMap())
+		t, err := t.Parse(content)
+		if err != nil {
+			return "", err
+		}
+
+		var tpl bytes.Buffer
+
+		err = t.Execute(&tpl, data)
+		if err != nil {
+			return "", err
+		}
+
+		content = tpl.String()
+		log.DebugF("Bundle script content:\n%s", content)
+	}
+
+	outFile, err := ioutil.TempFile(os.TempDir(), "*-detect-bundle.sh")
+	if err != nil {
+		return "", err
+	}
+
+	defer func() {
+		if err := outFile.Close(); err != nil {
+			log.ErrorF("Cannot close rendered detect_bundle.sh %s:%v", outFile.Name())
+		}
+	}()
+
+	if _, err = outFile.WriteString(content); err != nil {
+		return "", err
+	}
+
+	if err = outFile.Sync(); err != nil {
+		return "", err
+	}
+
+	if err = outFile.Chmod(0777); err != nil {
```

Comment:
```
	if err = outFile.Chmod(0777); err != nil {
```
Do we use 777 for scripts? I think for other files we use different permissions.